src/Dependencies/Updater.php

```diff
@@ -12,13 +12,62 @@ public function __construct(
         $this->dir = $dir;
     }
 
-    public function update()
+    public function update($dep = null)
     {
         $result = $this->loadWhippetFiles();
+
+        if ($result->isErr()) {
+            return $result;
+        }
+
+        if (is_null($dep)) {
+            return $this->updateAll();
+        }
+        return $this->updateDependency($dep);
+    }
+
+    private function updateDependency($dep)
```

Comment:
There seems to be a bunch of repeated code between `updateDependency` and `updateAll` (makes sense since they do the same thing).

It feels like there's a common function that can be extracted here. The fact that `updateDependency` is really really long and does multiple things should be a sign that some functions can be extracted.

Perhaps we need a function that accepts an array of dependencies and loops through them: `updateAll` passes the whole list of deps, and `updateDependency` (maybe rename to `updateSingle`?) passes an array containing the single dependency that it has found.